File: src/wickrcrypto/swig/java/com/wickr/crypto/tests/FingerprintTests.java
package com.wickr.crypto.tests;

import static org.junit.Assert.*;
import static org.hamcrest.CoreMatchers.*;
import org.junit.Test;
import org.junit.Before;
import com.wickr.crypto.*;
import java.io.*;
import java.util.*;

public class FingerprintTests
{

    protected Fingerprint testFingerprint;
        
    @Before
    public void setUp() throws UnsupportedEncodingException
    {
        byte[] identifier = CryptoEngine.digest("wickr".getBytes("UTF8"), null, Digest.sha256());
        ECKey sigKey = CryptoEngine.randEcKey(ECCurve.p521());
        Identity testIdentity = Identity.fromValues(IdentityType.IDENTITY_TYPE_ROOT, identifier, sigKey, null);
        this.testFingerprint = testIdentity.fingerprint();
    }

    @Test
    public void testEncodings() throws UnsupportedEncodingException
    {

        // Verify that encodings are working correctly
        String hexStringShort = testFingerprint.getHexString(FingerprintOutputType.FINGERPRINT_OUTPUT_SHORT);
        String base32StringShort = testFingerprint.getBase32String(FingerprintOutputType.FINGERPRINT_OUTPUT_SHORT);

        String hexStringLong = testFingerprint.getHexString(FingerprintOutputType.FINGERPRINT_OUTPUT_LONG);
        String base32StringLong = testFingerprint.getBase32String(FingerprintOutputType.FINGERPRINT_OUTPUT_LONG);

        assertTrue(hexStringLong.length() != testFingerprint.getData().length);
        assertTrue(hexStringLong.length() > hexStringShort.length());

        assertTrue(base32StringLong.length() != testFingerprint.getData().length);
        assertTrue(base32StringLong.length() > base32StringShort.length());

    }


}

File: src/wickrcrypto/swig/node/test/fingerprintTests.js
'use strict';

var expect = require('expect.js')
var wickrcrypto = require('../../../../../node/lib/wickrcrypto')

describe ("Fingerprint Encoding Tests", function() {

    var testFingerprint

    beforeEach(function() {
        var identifier = wickrcrypto.CryptoEngine.digest(Buffer.from("wickr"), null, wickrcrypto.Digest.sha256())
        var sigKey = wickrcrypto.CryptoEngine.randEcKey(wickrcrypto.ECCurve.p521())
        var testIdentity = wickrcrypto.Identity.fromValues(wickrcrypto.IDENTITY_TYPE_ROOT, identifier, sigKey, null)
        testFingerprint = testIdentity.fingerprint()
    })

    it ("should be able to encode fingerprints in various formats", function() {

        var hexStringShort = testFingerprint.getHex(wickrcrypto.FINGERPRINT_OUTPUT_SHORT).toString('utf8')
        var base32StringShort = testFingerprint.getB32(wickrcrypto.FINGERPRINT_OUTPUT_SHORT).toString('utf8')

        var hexStringLong = testFingerprint.getHex(wickrcrypto.FINGERPRINT_OUTPUT_LONG);
        var base32StringLong = testFingerprint.getB32(wickrcrypto.FINGERPRINT_OUTPUT_LONG);

        expect(hexStringLong.length != testFingerprint.data.length).to.be.true
        expect(hexStringLong.length > hexStringShort.length).to.be.true

        expect(base32StringLong.length != testFingerprint.data.length).to.be.true
        expect(base32StringLong.length > base32StringShort.length).to.be.true

    })

});